lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -889,8 +889,11 @@ std::vector<ConstString> SwiftLanguage::GetPossibleFormattersMatches(
 
   const bool check_cpp = false;
   const bool check_objc = false;
-  bool canBeSwiftDynamic = compiler_type.IsPossibleDynamicType(
-      nullptr, check_cpp, check_objc);
+  bool canBeSwiftDynamic =
+      compiler_type.IsPossibleDynamicType(nullptr, check_cpp, check_objc) ||
+      // A Swift @objc object in an Objective-C container comes in as (id).
+      (compiler_type.GetCanonicalType().GetTypeClass() ==
+       eTypeClassObjCObjectPointer);
```

Comment:
this `eTypeClassObjCObjectPointer` check occurs twice in the PR, should it be made less adhoc somehow? Should this check be moved into  `IsPossibleDynamicType`? Or are there cases where `eTypeClassObjCObjectPointer` is not considered possibly dynamic?

Reply:
> Should this check be moved into IsPossibleDynamicType?

It can't. It would have to be moved into *TypeSystemClang*::IsPossibleDynamicType() but in there we don't know that we are inside a Swift frame. And it makes no sense to return true for id pointers that aren't recognized by the ObjC runtime, since it couldn't resolve them.